base/threadingconstructs.jl

```diff
@@ -110,15 +111,73 @@ A barrier is placed at the end of the loop which waits for all tasks to finish
 execution.
 
 The `schedule` argument can be used to request a particular scheduling policy.
-The only currently supported value is `:static`, which creates one task per thread
-and divides the iterations equally among them. Specifying `:static` is an error
-if used from inside another `@threads` loop or from a thread other than 1.
+
+Except for `:static` scheduling, how the iterations are assigned to tasks, and how the tasks
+are assigned to the worker threads is undefined. The exact assignments can be different
+for each execution. The scheduling option is a hint. The loop body code (including any code
+transitively called from it) must not make assumptions about the distribution of iterations
+to tasks or the worker thread in which they are executed. The loop body for each iteration
+must be able to make forward progress independent of other iterations and be free from data
+races. As such, synchronizations across iterations may deadlock.
+
+For example, the above conditions imply that:
+
+- The lock taken in an iteration *must* be released within the same iteration.
+- Communicating between iterations using blocking primitives like `Channel`s is incorrect.
+- Write only to locations not shared across iterations (unless a lock or atomic operation is used).
+
+
+Schedule options are:
+- `:static` creates one task per thread and divides the iterations equally among
+            them, assigning each task specifically to each thread.
+            Specifying `:static` is an error if used from inside another `@threads` loop
+            or from a thread other than 1.
+- `:dynamic` will schedule iterations dynamically to available worker threads,
+            assuming that the workload for each iteration is uniform.
+
+Without the scheduler argument, the exact scheduling is unspecified; i.e. it may be
+different across Julia releases. Currently, the behavior is dependent on the calling thread.
+When called from thread 1 the default is `:static`. When called from other threads the loop
+will be executed without threading.
```

Comment:
Not sure how English works but isn't it easier to read if we have `,`s here?

```suggestion
When called from thread 1, the default is `:static`. When called from other threads, the loop
will be executed without threading.
```

Or maybe


```suggestion
The default is `:static` when called from thread 1. The loop will be executed without threading
when called from other threads.
```